### PR TITLE
Feature/#111 and #112  improve block detail and timetable in exchange mode

### PR DESCRIPTION
**Pull request by davidm1766**
review for #111 and #112

### CODE REVIEW

File: Backend/BlockChangesService.cs [hunk 1]
```diff
@@ -63,7 +68,7 @@ private async Task SetDoneStatus(BlockChangeRequest request)
         {
             if (request.Status == ExchangeStatus.Done)
             {
-                throw new ArgumentException("You cannot exchange requests which are already exchanged!");
+                await new Task(() => { throw new ArgumentException("You cannot exchange requests which are already exchanged!"); });
```
Comment (ikim23) [inline Backend/BlockChangesService.cs]: Why do you have to throw like this? Isn't simple throw enough?

File: Backend/BlockChangesService.cs [hunk 2]
```diff
@@ -11,12 +11,11 @@ namespace FRITeam.Swapify.Backend
...
-        private IMongoCollection<BlockChangeRequest> _blockChangesCollection => _database.GetCollection<BlockChangeRequest>(nameof(BlockChangeRequest));
+        private readonly IMongoCollection<BlockChangeRequest> _blockChangesCollection;
 
         public BlockChangesService(IMongoDatabase database)
         {
-            _database = database;
+            _blockChangesCollection = database.GetCollection<BlockChangeRequest>(nameof(BlockChangeRequest));
         }
 
         public async Task AddAsync(BlockChangeRequest entityToAdd)
```
Comment (ikim23) [inline Backend/BlockChangesService.cs]: Mongo should be able to generate Ids automatically...

File: Backend/BlockChangesService.cs (hunk 1 above)
Comment (davidm1766) [reply Backend/BlockChangesService.cs]: it is because of warning sonarlint...

File: Backend/BlockChangesService.cs (hunk 2 above)
Comment (davidm1766) [reply Backend/BlockChangesService.cs]: @ikim23 in whole app we assign ids manually

**Pull request merged**